### DISCUSSION TITLE
Add other validators to Server Actions

### DISCUSSION THREAD

**Discussion by huseeiin** [withastro/roadmap#993] [Proposal]
### Body

# Summary

You should add support for other validators like zod, yup, joi and especially typebox (which is what i want to use) and in addition, a custom function-based validator would be cool i.e. something like that:
```ts
const myAction = {
    input: (formData: Record<string, FormDataEntryValue>) => {
        if (!formData.email) return { error: "..." };

        return formData;
    },
};
```

# Background & Motivation

Not everyone uses zod. Plus its extremely slow compared to other competitors.

# Goals

Being able to choose the library you're most familiar with

# Example

`z.object` would be converted to `T.object` for typebox and so on for other validators
And please compile Typebox instead of using Check

**Comment by awhitford** [comment]
I'm keen to use [Valibot](https://valibot.dev/).  There is a [standard-schema](https://github.com/standard-schema/standard-schema) approach that could be leveraged.

**Reply by florian-lefebvre** [reply]
Standard Schema is too early atm I think, but I can't wait to see it be adopted before looking into using it in astro!

**Comment by florian-lefebvre** [comment]
I think you need to use `z.custom` for now, there are no near plans to support various validation libraries

**Reply by vixalien** [reply]
I was trying to migrate my app to astro, but ran into this issue. we already are using another validation and error scheme, and it's incompatible

**Comment by florian-lefebvre** [comment]
I encourage you to follow #1116 which includes this proposal and more
